documentation/src/main/asciidoc/topics/protobuf/personalshopper.proto

```diff
@@ -0,0 +1,12 @@
+message Person {
+    optional int32 id = 1;
+    required string name = 2;
```

Comment:
I'll not comment on all of the examples, but we recommend to not use the `required` field any more. ProtoStream is still based on proto2, however in proto3 all fields are optional so the `required` and `optional` fields are no longer available, so to aid migration it's better to use `optional` for all fields.

Reply:
The problem with optional is if the POJO is immutable and the fields are final, so the generated code must be able to build the object. Also scalar types are not nullable and 'required' fix that.
We might add a hint that this is deprecated, but I'm not sure when we have to migrate to proto 3

Reply:
> The problem with optional is if the POJO is immutable and the fields are final, so the generated code must be able to build the object

If `final` fields are desired on the POJO, then the correct way to build the object with Protostream is via a `@ProtoFactory` static method or constructor.

> Also scalar types are not nullable and 'required' fix that.

I don't understand the problem you're trying to fix here. Can you provide an example?

> We might add a hint that this is deprecated, but I'm not sure when we have to migrate to proto 3

It's more of a "best practice" as optional is far more flexible for allowing schemas to evolve in a backwards compatible manner, even with proto 2. 

I'm not sure what the current plan/timeline is for proto 3 support in Protostream .. @anistor?

Reply:
I've changed required -> optional